js/compiler.ts

```diff
@@ -407,14 +410,16 @@ window.compilerMain = function compilerMain(): void {
         }
       );
 
-    if (diagnostics.length > 0) {
-      host._logDiagnostics(diagnostics);
-      // The above _logDiagnostics calls os.exit(). The return is here just for
-      // clarity.
-      return;
-    }
+    const { emitSkipped } = emitResult;
+
+    const result: EmitResult = {
+      emitSkipped,
+      diagnostics: diagnostics.length
+        ? fromTypeScriptDiagnostic(diagnostics)
+        : undefined
+    };
 
-    postMessage(emitResult);
+    postMessage(result);
```

Comment:
Can you also remove the `_logDiagnostics()` method now?

Reply:
It was a bit more complex than that, because of the way the configuration is handled.  I rebased on master which included the compiler diagnostic changes and changed the logic of the compilation to fall through on diagnostics and return them and exit.